utils/buffer/src/buffer.rs

```diff
@@ -1,15 +1,41 @@
+// # Buffer
```

Comment:
aren't we documenting `BufferFromSystemMemory` here?

`Buffer` is on a different file